promql/engine.go

```diff
@@ -2108,7 +2114,12 @@ loop:
 				if floats == nil {
 					floats = getFPointSlice(16)
 				}
-				floats = append(floats, FPoint{T: t, F: f})
+				if n := len(floats); n < cap(floats) {
+					floats = floats[:n+1]
+					floats[n].T, floats[n].F = t, f
+				} else {
+					floats = append(floats, FPoint{T: t, F: f})
+				}
```

Comment:
I see the reason for the `cap` hack above for histograms (so that we can get a previously allocated histogram for re-use), but with float64, we don't need to re-use anything allocated.

Isn't `append` under the hood doing the same thing as you have implemented in the 1st branch of the `if` clause, provided the capacity is sufficient? Do you see any difference in the benchmark if you don't do this change?

Reply:
I thought we could provide a consistent way to resize the buffer for both floats and histograms, and also save an allocation for `FPoint`. However, benchmarks show 0% difference so reverting the change makes sense.